### DISCUSSION TITLE
"Bad" Mega 2560

### DISCUSSION THREAD

**Discussion by offron** [sanni/cartreader#663] [Need help]
So i have a "Bad" Mega 2560 one laying aroundthe one the wiki warns about.
Does that mean it wont work at all and do i need to get a new one, or could it work?

**Comment by sanni** [comment]
**Short answer:** 
It will work in the 5V setting but most likely will not work in the 3.3V setting. There are workarounds though but buying a working one is a quicker solution.

**Long answer:**
So far there have been three possible issues with the Mega 2560 when run out of spec at 3.3V and 16MHz like the Cart Reader does:

- brown-out detection set to high 
By default it is set to 2.7V which can be a bit too high when running at 3.3V especially since the Cart Reader needs to charge the 470uF cap too, so re-flashing the Mega Pro with different fuse settings to disable the brown out detection solves this.

- bad silicon quality
Running 3.3V at 16MHz is not something the factory tests for so some Mega Pros with worse than average silicon quality just can't run at 3.3V/16MHz no matter what. In this case you could still run 3.3V/8MHz by replacing the clock crystal and compiling with a custom 8Mhz Mega Pro hardware preset, dumping will take quite a lot longer then though.

- "bad" PCB design (bad as in for our use case)
This is the latest issue the Wiki warns about, there is a design out there that for whatever reason needs 3.3V at the 3.3V pin of the Arduino to run while on the older design the 3.3V pin is not needed at all. (If the Mega Pro is run at 5V the 3.3V at the 3.3V pin will be provided by the AMS1117 3.3 on its PCB. But when run at 3.3V that AMS1117 will only output 1.8V.)
There has been a work around posted here that just connects 3.3V from the Cart Reader to the 3.3V pin of the Mega Pro but that will short-circuit the input and output of the AMS1117 3.3 on the Mega Pro PCB and short-circuits on a voltage regulator are less than ideal, so I removed that fix again. Someone could probably figure out a proper fix though.

**Reply by nmfernan** [reply]
Hey, sanni

So is option one a known fix? The 'brown-out' bit, that is. It makes sense on paper, but wasn't sure if this has been a widespread fix.

I'm having a heck of a time trying to find the right Mega not on Aliexpress (I don't wanna wait a zillion years to build it). I'm looking at this option specifically: https://www.amazon.com/Gowoops-ATMEGA2560-16AU-headers-Compatible-Mega2560/dp/B07Y9P4JL2/ref=sr_1_1?crid=2MQWJGZ82PK82&keywords=2560+pro+16u&qid=1672661972&sprefix=2560+pro+16u%2Caps%2C548&sr=8-1#

**Reply by sanni** [reply]
The brown-out fix should apply to cases where you have the known good PCB design and it still has problems with booting or read errors on 3.3V.

The pictured Mega Pro in the Amazon link looks good. Order it and if the Mega Pro is not the same as in the picture you can return it easily.

**Reply by nmfernan** [reply]
Thanks for the super quick reply!

So if it isn't the right version, any idea where I can get a ATMega2560-16U version? I've been trying pretty hard to find one and have no luck

Is it the ATMega2560-16AU specifically, or the different configuration of resistors shown in the "bad" image?
